### PR TITLE
2.27.0 2.22.0

### PR DESCRIPTION
**Pull request by rrthomas**

### CODE REVIEW

File: examples/2-subdivision-number-in-header.tex [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+\begin{document}
+
+
+\begin{english}
+\date{}
+\title{Subdivision number inside the header}
+\maketitle
+\begin{abstract}
+  This file show how to use the \verb+\doinsidethislinehook+
```
Comment (rrthomas) [inline examples/2-subdivision-number-in-header.tex]: "This file shows how to use the…"

File: examples/2-subdivision-number-in-header.tex [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+\begin{document}
+
+
+\begin{english}
+\date{}
+\title{Subdivision number inside the header}
+\maketitle
+\begin{abstract}
+  This file show how to use the \verb+\doinsidethislinehook+ 
+  command to use \verb+\markboth+ to add verse (sentence) number inside the headers.
```
Comment (rrthomas) [inline examples/2-subdivision-number-in-header.tex]: numbers

File: examples/2-subdivision-number-in-header.tex [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+  
+  We can't call \verb+\markboth+ directly inside the \verb+pstart+…\verb+\pend+, because it creates trouble with vertical spacing.
+ 
+  We show two examples: the first one add directly the verse number inside the source code, while the second use a counter to determine it automatically.
+
+  To understand this example, you must be familiar with \LaTeX\ marking mechanism (\verb+\markboth+) and \emph{fancyhdr} package.
```
Comment (rrthomas) [inline examples/2-subdivision-number-in-header.tex]: the \emph{fancyhdr} package.

File: examples/2-subdivision-number-in-header.tex [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+\end{english}
+
+\begin{english}
+\section{Manuel subdivision number}
+
+\begin{abstract}
+ In this example, the verse (sentence) number is typeset directly in the source code.
+ We wrap it in a \verb+\versenumber+ macro, which:
+ \begin{itemize}
+   \item typesets it with a bold fontface, following by a dot;
```
Comment (rrthomas) [inline examples/2-subdivision-number-in-header.tex]: followed by

File: examples/2-subdivision-number-in-header.tex [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+   \item use \verb+\doinsidelinehook+ to call correctly \verb+\markboth+.
```
Comment (rrthomas) [inline examples/2-subdivision-number-in-header.tex]: uses

File: examples/2-subdivision-number-in-header.tex [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+\versenumber{87} Vestibulum erat ante, maximus id pulvinar vitae, sagittis sed odio.
+\pend
+\endnumbering
+\newpage
+
+\begin{english}
+\section{Automatic verse number}
+
+\begin{abstract}
+ In this example, the verse (sentence) number is not typeset directly, but store in a \LaTeX\ counter (which name is \verb+verse+).
```
Comment (rrthomas) [inline examples/2-subdivision-number-in-header.tex]: stored in…(whose name is…)

File: examples/2-subdivision-number-in-header.tex [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+ The \verb+\autoversenumber+ macro:
+ \begin{itemize}
+   \item increases the counter;
+   \item typesets it with a bold fontface, following by a dot;
```
Comment (rrthomas) [inline examples/2-subdivision-number-in-header.tex]: fontface → face

Comment (rrthomas) [reply examples/2-subdivision-number-in-header.tex]: followed by

File: examples/2-subdivision-number-in-header.tex [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+   \item use \verb+\doinsidelinehook+ to call correctly \verb+\markboth+.
+ \end{itemize}
+
+ The main problem is that, because of \emph{reledmac} mechanism,  the lines are typeset after the \verb+\pend+, 
+ at a time when the \verb+verse+ counter is already set to its maximal value.
+ Consequently, with hack, the \verb+\markboth+ macro won't have set the good value to the marker.
```
Comment (rrthomas) [inline examples/2-subdivision-number-in-header.tex]: with the hack, the \verb+\markbox+ macro won't have set the marker to the correct value.

File: examples/2-subdivision-number-in-header.tex [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+
+ However, we can solve this issue, because the argument of \verb+doinsidethislinehook+ is written in a auxiliary file
+ at the time where  it is read.
```
Comment (rrthomas) [inline examples/2-subdivision-number-in-header.tex]: at the time where  it is read → when it is read.

File: examples/2-subdivision-number-in-header.tex [hunk 10]
```diff
@@ -0,0 +1,282 @@
...
+
+Nevertheless, to write the correct value of the counter in the auxiliary file, we need to do some hacks with \TeX\ expansion's mechanism.
```
Comment (rrthomas) [inline examples/2-subdivision-number-in-header.tex]: with \TeX\'s expansion mechanism.

File: examples/2-subdivision-number-in-header.tex [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+
+\end{abstract}
+\end{english}
+
+\newcounter{verse}
+\newcommand{\autoversenumber}{%
+  \refstepcounter{verse}%Add 1 to the counter
+  \edef\tmp{\noexpand\markboth{\the\value{verse}}{\the\value{verse}}}%\tmp will be expanded to \markboth{<verse-value>}{<verse-value}>
+  \expandafter\doinsidethislinehook\expandafter{\tmp}%First, we expand \tmp to  the value, ans then, pass the result as argument to \doinsidelinehook
```
Comment (rrthomas) [inline examples/2-subdivision-number-in-header.tex]: ans then, pass → and then pass

File: examples/2-subdivision-number-in-header.tex [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+  \textbf{\theverse}.\space%Typeset the verse counter value in bold, following by a dot and then a space.
```
Comment (rrthomas) [inline examples/2-subdivision-number-in-header.tex]: followed by

File: reledpar.dtx [hunk 13]
```diff
@@ -641,11 +646,14 @@
...
-% \cs{resumenumbering}. The important thing to remember is that you have 
-% to use \cs{endnumbering} in a right column before every 
-% \cs{endnumbering} in single-column text or in parallel with 
-% \cs{endnumbering} in a left-column text (the latter is the case when a 
-% document or a section ends in a two-columns mode). Without this, you will 
+% \cs{resumenumbering}. 
+% 
+% You must use \cs{endnumbering} in a right column in the following cases:
+% \begin{itemize}
+%  \item there is also a \cs{endnumbering} in the left column, because the line numbering ends with a two columns mode (for ex. at the end of a document);
```
Comment (rrthomas) [inline reledpar.dtx]: for ex. → e.g.

File: reledpar.dtx [hunk 14]
```diff
@@ -641,11 +646,14 @@
...
+%  \item there will be a \cs{endnumbering} in a text after the columns, in a single column mode, because the ligne numbering finish with single column mode text.
```
Comment (rrthomas) [inline reledpar.dtx]: ligne numbering finish → line numbering finishes

**Comment by maieul** [comment]
I will correct in the release branch, and then merge the release branch monday on master, and then merge master on english2. 

We should not merge directly to english2.

**Comment by maieul** [comment]
Thanks, I will do the release process now.

**Pull request closed without merging**